pkg/daemon/ceph/osd/remove.go

```diff
@@ -116,6 +83,39 @@ func removeOSD(clusterdContext *clusterd.Context, clusterInfo *client.ClusterInf
 		logger.Errorf("failed to exclude osd.%d out of the crush map. %v", osdID, err)
 	}
 
+	// Check we can remove the OSD
+	// Loop forever until the osd is safe-to-destroy
+	for {
+		isSafeToDestroy, err := client.OsdSafeToDestroy(clusterdContext, clusterInfo, osdID)
+		if err != nil {
+			// If we want to force remove the OSD and there was an error let's break outside of
+			// the loop and proceed with the OSD removal
+			if forceOSDRemoval {
```

Comment:
Not related to this PR, but why do we have `forceOSDRemoval` check after checking `OsdSafeToDestroy` ?
If `forceOSDRemoval` is true, then can `OsdSafeToDestroy` check be skipped completely?

Reply:
right, that was the intention for adding `forceOsdRemoval`

Reply:
just continuing the offline discussion I had with @sp98  here,

 we should skip `OsdSafeToDestroy` check when `forceOSDRemoval` passed since the purpose of the `forceOSDRemoval`  flag is that that we don't care of data loss and want to remove osd forcefully. But, if we pass more than one osd id if the removal job that could lead to data loss(https://github.com/rook/rook/pull/9230 more context here). So, what should be the right way to go with @travisn @BlaineEXE